compiler/rustc_passes/src/dead.rs

```diff
@@ -142,7 +142,11 @@ impl<'tcx> MarkSymbolVisitor<'tcx> {
                 self.insert_def_id(def.non_enum_variant().fields[index].did);
             }
             ty::Tuple(..) => {}
-            _ => span_bug!(lhs.span, "named field access on non-ADT"),
+            ty => {
```

Comment:
I don't really like my change. It feels more like I'm treating symptoms instead of a root cause.

Alternatively can check `references_errors()` and keep the `span_bug` (which still wouldn't address the point above).

Reply:
Can you change the callsite to check the typeck results if a `opt_field_index` has been recorded for this expr? Otherwise, delay a bug. This can remain a span bug then.

Reply:
This is consistent with what I did in https://github.com/rust-lang/rust/commit/d442c015d6d688460a182947e797f4bf69876044#diff-945cde608e29cea98a59d63e87bcac74d84a2390a1cfa71b1c89d08007789850R527

Reply:
Much better, thanks!